internal/cloud/awscloud/secure-instance.go

```diff
@@ -436,3 +422,35 @@ func (a *AWS) deleteSGIfExists(si *SecureInstance) error {
 	}
 	return err
 }
+
+func (a *AWS) createFleet(input *ec2.CreateFleetInput) (*ec2.CreateFleetOutput, error) {
+	createFleetOutput, err := a.ec2.CreateFleet(input)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create fleet: %w", err)
+	}
+
+	if len(createFleetOutput.Errors) > 0 && createFleetOutput.Errors[0].ErrorCode == aws.String("UnfillableCapacity") {
+		logrus.Warn("Received UnfillableCapacity from CreateFleet, retrying CreateFleet with OnDemand instance")
+		input.SpotOptions = nil
+		createFleetOutput, err = a.ec2.CreateFleet(input)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create fleet: %w", err)
```

Comment:
Can we change the error string, so we know which one failed?
```suggestion
		return nil, fmt.Errorf("Unable to create OnDemand fleet: %w", err)
```